apps/asap-cli/src/scripts/algolia/process-productivity-metrics.ts

```diff
@@ -0,0 +1,230 @@
+    averageMin: isInteger
+      ? Math.ceil(inferiorLimit)
+      : roundToTwoDecimals(inferiorLimit, 'ceil'),
+    averageMax: isInteger
+      ? Math.floor(superiorLimit)
+      : roundToTwoDecimals(superiorLimit, 'floor'),
+    aboveAverageMin: isInteger
+      ? Math.ceil(superiorLimit)
+      : roundToTwoDecimals(superiorLimit, 'ceil'),
+    aboveAverageMax: Math.max(...data),
+  };
+};
+
+type Hit = {
+  objectID: string;
+};
+
+type UserProductivityHit = Hit & {
+  asapOutput: number;
+  asapPublicOutput: number;
+  ratio: string;
+};
+
+type TeamProductivityHit = Hit & {
+  [documentType in TeamOutputDocumentType]: number;
+};
+
+export const getAllHits = async <T>(
+  getPaginatedHits: (page: number) => Promise<SearchResponse<T>>,
+): Promise<T[]> => {
+  let page = 0;
+  let totalPages = 0;
+  let allHits: T[] = [];
+
+  /* eslint-disable no-await-in-loop */
+  do {
+    const result = await getPaginatedHits(page);
+    allHits = allHits.concat(result.hits);
+    if (totalPages === 0) {
+      totalPages = result.nbPages;
+    }
+    page += 1;
+  } while (page < totalPages);
+  /* eslint-enable no-await-in-loop */
+
+  return allHits;
+};
+
+export const deletePreviousObjects = async (
+  index: SearchIndex,
+  type: 'user-productivity' | 'team-productivity',
+) => {
+  const previous = await index.search('', {
+    filters: `__meta.type:"${type}-metrics"`,
+  });
+  const objectIDs = previous.hits.map(({ objectID }) => objectID);
+  await index.deleteObjects(objectIDs);
+};
+
+export const processUserProductivityMetrics = async (index: SearchIndex) => {
+  const type = 'user-productivity' as const;
+  await deletePreviousObjects(index, type);
+
+  timeRanges.forEach(async (range) => {
+    const getPaginatedHits = (page: number) =>
+      index.search<UserProductivityHit>('', {
+        filters: `__meta.range:"${range}" AND (__meta.type:"${type}")`,
+        attributesToRetrieve: ['asapOutput', 'asapPublicOutput', 'ratio'],
+        page,
+        hitsPerPage: 50,
+      });
+
+    const userProductivityHits =
+      await getAllHits<UserProductivityHit>(getPaginatedHits);
+
+    const fields = ['asapOutput', 'asapPublicOutput', 'ratio'];
+
+    const userMetrics = fields.reduce(
+      (metrics, field) => {
+        if (field === 'ratio') {
+          return {
+            ...metrics,
+            ratio: getBellCurveMetrics(
+              userProductivityHits.map((hit) => parseFloat(hit.ratio)),
+              false,
+            ),
+          };
+        }
+
+        return {
+          ...metrics,
+          [field]: getBellCurveMetrics(
+            userProductivityHits.map(
+              (hit) => hit[field as 'asapOutput' | 'asapPublicOutput'],
+            ),
+          ),
+        };
+      },
+      {} as Record<string, PerformanceMetrics>,
+    );
+
+    await index.saveObject(
+      {
+        ...userMetrics,
+        __meta: {
+          range,
+          type: `${type}-metrics`,
```

Comment:
yeah... are you sure about that `-metrics` suffix? It took a me a while to understand what you were saving here. Maybe this ball curve thing is a better way as thats essentially what these records are, whereas technically speaking everything we have in Algolia for the analytics is a metric, no?